pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,645 @@
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, tlsAuthSecret, func() error {
+		tlsAuthSecret.Type = corev1.SecretTypeOpaque
+		tlsAuthSecret.Data = v.secrets.TLSAuth.Data
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, dhSecret, func() error {
+		dhSecret.Type = corev1.SecretTypeOpaque
+		dhSecret.Data = v.secrets.DH.Data
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, networkPolicy, func() error {
+		networkPolicy.ObjectMeta.Annotations = map[string]string{
+			v1beta1constants.GardenerDescription: "Allows Egress from pods labeled with 'networking.gardener.cloud/to-vpn-seed-server=allowed' to talk to's Kubernetes API Server.",
+		}
+		networkPolicy.Spec = networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					v1beta1constants.GardenRole: "controlplane",
+					v1beta1constants.LabelApp:   "vpn-seed-server",
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.GardenRole: "controlplane",
+									v1beta1constants.LabelApp:   "kubernetes",
+									v1beta1constants.LabelRole:  "apiserver",
+								},
+							},
+						},
+					},
+				},
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							// we don't want to modify existing labels on the istio namespace
+							NamespaceSelector: &metav1.LabelSelector{},
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.LabelApp: "istio-ingressgateway",
+								},
+							},
+						},
+					},
+				},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.GardenRole: "controlplane",
+									v1beta1constants.LabelApp:   "kubernetes",
+									v1beta1constants.LabelRole:  "apiserver",
+								},
+							},
+						},
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, deployment, func() error {
+		maxSurge := intstr.FromInt(100)
+		maxUnavailable := intstr.FromInt(0)
+		deployment.Labels = map[string]string{
+			v1beta1constants.GardenRole:                           "controlplane",
+			v1beta1constants.LabelApp:                             DeploymentName,
+			v1beta1constants.LabelNetworkPolicyFromShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+		}
+		deployment.Spec = appsv1.DeploymentSpec{
+			MinReadySeconds:      30,
```

Comment:
Out of curiosity: Is this field set consciously? If yes, why?

Reply:
No, it will be removed.